eyeshade/workers/reports.js

```diff
@@ -109,6 +118,48 @@ const daily = async (debug, runtime) => {
   debug('daily', 'running again ' + moment(tomorrow).fromNow())
 }
 
+exports.freezeOldSurveyors = freezeOldSurveyors
+
+async function freezeOldSurveyors (olderThanDays, anchorTime, surveyors) {
+  // in seconds
+  const nDays = DAY_SECONDS * olderThanDays
+  const nDaysAgo = (anchorTime - nDays)
+  const nDaysNumber = nDaysAgo * 1000
+  const where = {
+    frozen: false,
+    surveyorType: 'contribution'
+  }
+  const data = {
+    $set: {
+      frozen: true
+    }
+  }
+  const allNonFrozenSurveyors = await surveyors.find(where)
+  const freezeUpdates = allNonFrozenSurveyors.map(surveyorExpirer)
+  await Promise.all(freezeUpdates)
+    .then(() => freezeUpdates.length)
+
+  function surveyorExpirer (surveyor) {
+    const {
+      _id,
+      timestamp
+    } = surveyor
+    const where = {
+      _id
+    }
+    if (!timestamp) {
+      // bad surveyor
+      return
+    }
+    const tsmp = +createdTimestamp(_id)
```

Comment:
is the `+` here intentional?

Reply:
yes. it turns the value into a number. just did some tests and found that it is an unnecessary step as you can compare a date object to a number

Reply:
... so i will remove